contracts/multisig/src/contract.rs

```diff
@@ -132,20 +133,32 @@ pub mod execute {
 
         let signer = info.sender.clone().into();
 
-        let (signature, state) = sign(deps.storage, &session, &key, signer, signature)?;
+        let signature = sign(
+            deps.storage,
+            &mut session,
+            &key,
+            signer,
+            signature,
+            env.block.height,
+        )?;
 
         let event = Event::SignatureSubmitted {
             session_id,
             participant: info.sender,
             signature,
         };
 
-        if state == MultisigState::Completed {
-            Ok(Response::new()
-                .add_event(event.into())
-                .add_event(Event::SigningCompleted { session_id }.into()))
-        } else {
-            Ok(Response::new().add_event(event.into()))
+        match session.state {
+            MultisigState::Pending => Ok(Response::new().add_event(event.into())),
+            MultisigState::Completed { completed_at } => {
+                Ok(Response::new().add_event(event.into()).add_event(
+                    Event::SigningCompleted {
+                        session_id,
+                        completed_at,
+                    }
+                    .into(),
+                ))
+            }
         }
```

Comment:
this is going to fire with every late signature

Reply:
I did this on purpose, I assumed either way it would be helpful to get back an event indicating the session was completed. I added the `completed_at` to the event so it can be determined if the completion was in the past. Thoughts?

Reply:
I think it's more confusing than helpful. Events don't get shown as a response when you call the contract from the cli for example, rather they get returned to specific subscription which could make the reasonable assumption that every signing can only completed once. If you want to return an event, it should be a different one, like `SignedAfterCompletion`